src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -126,6 +136,9 @@ namespace hit {
         sk = unmarshalBinarySecretKey(secret_key_stream);
         galois_keys = unmarshalBinaryRotationKeys(galois_key_stream);
         relin_keys = unmarshalBinaryRelinearizationKey(relin_key_stream);
+        if (context->btp_params.has_value()) {
```

Comment:
NitPick: There seems to be lots of shared logic in this constructor and the other deserializing constructor (directly above) which could be abstracted out.

Reply:
Created a new helper function.